internal/graph/check.go

```diff
@@ -353,13 +353,19 @@ func filterForFoundMemberResource(resourceRelation *core.RelationReference, reso
 
 	for index, resourceID := range resourceIds {
 		if subject.ObjectId == resourceID {
-			return []string{resourceID}, append(resourceIds[:index], resourceIds[index+1:]...)
+			return []string{resourceID}, removeIndexFromSlice(resourceIds, index)
 		}
 	}
 
 	return nil, resourceIds
 }
 
+func removeIndexFromSlice[T any](s []T, index int) []T {
```

Comment:
You can probably use the trick where you move the last element onto the one you want to remove, and then return a truncated slice. This will prevent any allocs. Consider using a set though.

Reply:
We can't reuse the slice, as the race detector complains about us modifying it. I had that originally, but had to change to the copy to fix it

Reply:
Ah yeah I see in one of the calls to `filterForFoundMemberResource` we're actually passing a shared slice.